crates/sui-types/src/crypto.rs

```diff
@@ -313,6 +313,20 @@ where
     get_key_pair_from_rng(&mut OsRng)
 }
 
+/// Wrapper function to return SuiKeypair based on key scheme string
+pub fn random_key_pair_by_type(key_scheme: Option<String>) -> (SuiAddress, SuiKeyPair) {
+    match key_scheme.as_deref() {
+        Some("secp256k1") => {
+            let (addr, key_pair): (_, Secp256k1KeyPair) = get_key_pair();
+            (addr, SuiKeyPair::Secp256k1SuiKeyPair(key_pair))
+        }
+        Some(_) | None => {
```

Comment:
interpret any string that is not exactly "secp256k1" to use Ed25519, we can enforce strict matching here, open to suggestions

Reply:
Good point, I'd suggest strict matching plz, blind selections have caused many issues in the past.

Reply:
strict matching will require some refactoring on this PR